akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2.scala

```diff
@@ -100,6 +104,15 @@ private[http] final class Http2Ext(private val config: Config)(implicit val syst
       }.to(Sink.ignore).run()
   }
 
+  private def prepareServerAttributes(settings: ServerSettings, incoming: Tcp.IncomingConnection) = {
+    val attrs = Http.prepareAttributes(settings, incoming)
+    if (telemetry == NoOpTelemetry) attrs
+    else {
+      // transfer attributes allowing context propagation from serverBinding interceptor to request-response interceptor
```

Comment:
That's really subtle (and potentially dangerous depending on the concrete set of attribute we might set on `IncomingConnection.flow`). This tight coupling to implementations of `IncomingConnection` sounds like a viability if we ever have to change details in `IncomingConnection`.

Reply:
Yeah, I agree. This is basically doing the exact same thing that cinnamon does today, I tried hard coming up with another solution for passing context between the right places but in the end I gave up and went with this.

Reply:
We have now reverted to do less HTTP/2 specifics in Cinnamon and rely more on the existing HTTP/1 instrumentation, so doing this some other way is even less of an option without lots of extra work now.